### DISCUSSION TITLE
How to implement ExportEndpoint functionality for manipulating the wsdl without ServiceHostFactory using CoreWCF in NET6?

### DISCUSSION THREAD

**Discussion by arunrkexperion** [CoreWCF/CoreWCF#1154] [General]
Hi CoreWCF Team. We are having an existing WCF implementation which uses the ServiceHostFactory. Seems the same is not available in CoreWCF. We are trying to achieve  this 
"public class FlatWsdl : IWsdlExportExtension, IEndpointBehavior
    {
        /// <summary>
        /// Writes custom Web Services Description Language (WSDL) elements into the generated WSDL for an endpoint.
        /// </summary>
        /// <param name="exporter">The <see cref="T:System.ServiceModel.Description.WsdlExporter"/> that exports the endpoint information.</param><param name="context">Provides mappings from exported WSDL elements to the endpoint description.</param>
        public void ExportEndpoint(WsdlExporter exporter, WsdlEndpointConversionContext context)
        {
        }
 "
ExportEndpoint functionality for manipulating the wsdl. Could you please let us know if you have an alternate work around for achieving this? Thanks.

**Comment by mconnew** [comment]
The same mechanism works for CoreWCF as on WCF. If you add your IEndpointBehavior to your endpoint it automatically checks every configured endpoint behavior to see if it implements IWsdlExportExtension and if it does, uses it. An easy way to do this would be to create an IServiceBehavior which adds your endpoint behavior to each relevant endpoint, then add it to DI as a singleton (`services.AddSingleton<IServiceBehavior, MyServiceBehaviorType>()`) and it will automatically be picked up.
